api/grpc/mpi/v1/file_stream_test.go

```diff
@@ -0,0 +1,512 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package v1_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+	"github.com/nginx/agent/v3/api/grpc/mpi/v1/v1fakes"
+)
+
+var letters = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+//nolint:gosec
+func randBytes(n int) []byte {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+
+	return b
+}
+
+//nolint:gocognit,revive
+func TestSendChunkedFile(t *testing.T) {
+	for desc, td := range map[string]struct {
+		clientFunc  func(cl *v1fakes.FakeFileService_GetFileStreamServer)
+		header      v1.FileDataChunk_Header
+		expectedErr error
+		content     []byte
+	}{
+		"empty": {
```

Comment:
nit: I believe this project has a test naming convention, want to prefix these test cases with "Test 1:", "Test 2:", etc